app/Http/Controllers/RoomController.php

```diff
@@ -0,0 +1,102 @@
+<?php
+
+namespace App\Http\Controllers;
+
+use App\Models\Room;
+use Illuminate\Http\Request;
+
+class RoomController extends Controller
+{
+    /**
+     * Display a listing of the resource.
+     *
+     * @return \Illuminate\Http\Response
+     */
+    public function index()
+    {
+        return inertia('Admin/Rooms/Index', [
+            'rooms' => Room::all(),
+        ]);
+    }
+
+    /**
+     * Show the form for creating a new resource.
+     *
+     * @return \Illuminate\Http\Response
+     */
+    // public function create()
+    // {
+    //     //
+    // }
+
+    /**
+     * Store a newly created resource in storage.
+     *
+     * @param  \Illuminate\Http\Request  $request
+     * @return \Illuminate\Http\Response
+     */
+    public function store(Request $request)
+    {
+
+        $request->validateWithBag('createRoom', [
+            'name' => ['required', 'string', 'max:255'],
+            'number' => ['required', 'string', 'max:255'],
+            'floor' => ['required', 'integer'],
+            'building' => ['required', 'string', 'max:255'],
+        ]);
+
+        Room::create([
+            'name' => $request->name,
+            'number' => $request->number,
+            'floor' => $request->floor,
+            'building' => $request->building
+        ]);
+
+        return back()->with('flash', ['we good']);
```

Comment:
```suggestion
        return back();
```

We can remove the flash since its never used in the frontend.